js/src/api/util/decode.js

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { isHex } from './types';
+
+import Func from '../../abi/spec/function';
+
+const CREATE_METHOD = '60606040';
+
+export function decodeCallData (data) {
+  if (!data || !data.length) {
+    return {};
+  }
+
+  if (!isHex(data)) {
+    throw new Error('Input to decodeCallData should be a hex value');
+  }
+
+  if (data.substr(0, 2) === '0x') {
+    return decodeCallData(data.slice(2));
+  } else if (data.length < 8) {
+    throw new Error('Input to decodeCallData should be method signature + data');
+  }
+
+  const signature = data.substr(0, 8);
+  const paramdata = data.substr(8);
+
+  if ((signature !== CREATE_METHOD) && paramdata.length % 64 !== 0) {
+    throw new Error(`Parameter length in decodeCallData not a multiple of 64 characters, ${paramdata.length}`);
+  }
+
+  return {
+    signature: `0x${signature}`,
+    paramdata: `0x${paramdata}`
+  };
+}
+
+export function decodeMethodInput (methodAbi, paramdata) {
+  if (!methodAbi) {
+    throw new Error('decodeMethodInput should receive valid method-specific ABI');
+  } else if (!paramdata) {
+    throw new Error('decodeMethodInput should receive valid parameter input data');
+  } else if (!isHex(paramdata)) {
+    throw new Error('Input to decodeMethodInput should be a hex value');
+  } else if (paramdata.substr(0, 2) === '0x') {
+    return decodeMethodInput(methodAbi, paramdata.slice(2));
+  } else if (paramdata.length % 64 !== 0) {
+    throw new Error('Parameter length in decodeMethodInput not a multiple of 64 characters');
+  }
+
+  return new Func(methodAbi).decodeInput(paramdata).map((decoded) => decoded.value);
+}
+
+// takes a method in form name(..., types) and returns the interred abi definition
+export function methodToAbi (method) {
```

Comment:
The `indexOf` feels too brittle. This function should validate the syntax of `method` properly (right now, it will already give a false positive with spaces inside `method`).

If this is a standard format, as @jacogr told me, there should be an independent module for validating!